lensless/recon/trainable_recon.py

```diff
@@ -58,29 +66,87 @@ def __init__(self, psf, dtype=None, n_iter=1, **kwargs):
                 Data type to use for optimization.
             n_iter : int
                 Number of iterations for unrolled algorithm.
+            post_process : :py:class:`function` or :py:class:`~torch.nn.Module`, optional
+                If :py:class:`function` : Function to apply to the image estimate after the whole ADMM algorithm. Its input most be (image to process, noise_level), where noise_level is a learnable parameter. If it include aditional learnable parameters, they will not be added to the parameter list of the algorithm. To allow for traning, the function must be autograd compatible.
+                If :py:class:`~torch.nn.Module` : A DruNet like network. The network will be included as a submodule of the algorithm and its parameters will be added to the parameter list of the algorithm. If this isn't intended behavior, set requires_grad=False.
         """
         assert isinstance(psf, torch.Tensor), "PSF must be a torch.Tensor"
         super(TrainableReconstructionAlgorithm, self).__init__(
             psf, dtype=dtype, n_iter=n_iter, **kwargs
         )
 
-    @abc.abstractmethod
+        # pre processing
```

Comment:
setup code for pre_process and post_process are very similar, could you define a class method with the following API:
```
def _set_process_block(self, process):
    process_model = None
    process = None
    process_param = None
    ...
    return process_model, process, process_param


# inside constructor
self.pre_process_model, self.pre_process, self.pre_process_param = self._set_process_block(pre_process)
self.post_process_model, self.post_process, self.post_process_param = self._set_process_block(post_process)
```